Sources/KeyboardSupport/KeyboardAutoNavigator.swift

```diff
@@ -0,0 +1,221 @@
+//
+//  KeyboardAutoNavigator.swift
+//  KeyboardSupport-iOS
+//
+//  Created by John Davis on 12/3/18.
+//  Copyright © 2018 Bottle Rocket. All rights reserved.
+//
+
+import Foundation
+
+/// Contains callbacks for `KeyboardAutoNavigator` navigation events.
+public protocol KeyboardAutoNavigatorDelegate: class {
+    func keyboardNavigatorDidTapBack(_ navigator: KeyboardAutoNavigator)
+    func keyboardNavigatorDidTapNext(_ navigator: KeyboardAutoNavigator)
+    func keyboardNavigatorDidTapDone(_ navigator: KeyboardAutoNavigator)
+}
+
+/// Handles navigating between text fields in a containing view hierarchy.
+open class KeyboardAutoNavigator: KeyboardNavigating {
+    
+    /// AutoPilot is a collection of static functions that enable navigating between UITextInputViews in a view hierarchy
+    public enum AutoPilot {
```

Comment:
I think AutoPilot could be better represented as a `struct` since it does not have any cases.